### ISSUE TITLE
Feautre request : scripting : new note with filename indepentent of headline

### ISSUE THREAD

**Issue by bradmont** [pbek/QOwnNotes#1046]
Hi,

I'm writing a simple script to create a new blog post template for the [hugo](http://gohugo.io) static blog engine. Hugo uses markdown files with metadata at the beginning (see below). The script is writing this forematter for me, but taking the filename from the first line means all files are named "---.md". It's not too much work to rename them, but it would be better for the script to do it. Is there a way to do this? I'm not seeing it in the documentation.

An example of the forematter the script is generating is:
```
---
title: "I love qownnotes"
date: 2018-10-17T11:46:31-4:00
draft:true
---
```

If there isn't currently a way to do this, an overloaded createNote function, like createNote (content, fileName) would do nicely.

Thanks!

**Comment by pbek** [comment]
Would http://docs.qownnotes.org/en/develop/scripting/README.html#handlenotetextfilenamehook be enough for your script?

**Comment by bradmont** [comment]
Oh, yes it does! I saw it while searching the docs but I guess I didn't understand it. The first of the two examples doesn't actually change the name and I guess I didn't check the second one, which does.

I had to add a property to the QtObject to track whether the note is newly created by my script and only modify the name in that case, otherwise it would really wreak havoc with all of my other notes, haha!

Thanks for the help!
